### DISCUSSION TITLE
Synology - Docker - Kernal

### DISCUSSION THREAD

**Discussion by sn0west** [firefly-iii/firefly-iii#8122] [Firefly III]
Dear all,

I'm not able to install the latest Firefly version. The Docker container can't start, it always restart evry 10-20 sekonds.
I had a similar problem with the newest Nextxcloud Docker version. I had to change the docker file from "php:8.0-apache-bullseye zu  FROM php:8.0-apache-buster" and the create the new image, and it worked fine.

Yesterday I installed firefly version 5.7.9 and it worked fine. I think it is a similar problem, that firefly is no compatible with my synology kernal version. I found the docker generate file "[](https://dev.azure.com/firefly-iii/_git/BaseImage?path=/generate-docker-files.php)". So I can create my own docker image file. Can anyone tell me please what exactly I have to do do, to make it work? I have Synology DSM 6.2.4.

What I would do is, remove the unnecesery lines and leave it like this.
`$files = [
    'apache' => [
    sprintf('https://raw.githubusercontent.com/docker-library/php/master/%s/buster/apache/Dockerfile', PHP_TO_USE),
    sprintf('https://raw.githubusercontent.com/docker-library/php/master/%s/buster/apache/docker-php-entrypoint', PHP_TO_USE),
    sprintf('https://raw.githubusercontent.com/docker-library/php/master/%s/buster/apache/apache2-foreground', PHP_TO_USE),
    sprintf('https://raw.githubusercontent.com/docker-library/php/master/%s/buster/apache/docker-php-ext-configure', PHP_TO_USE),
    sprintf('https://raw.githubusercontent.com/docker-library/php/master/%s/buster/apache/docker-php-ext-enable', PHP_TO_USE),
    sprintf('https://raw.githubusercontent.com/docker-library/php/master/%s/buster/apache/docker-php-ext-install', PHP_TO_USE),
    sprintf('https://raw.githubusercontent.com/docker-library/php/master/%s/buster/apache/docker-php-source', PHP_TO_USE),
    ],
];`

I don't know the difference between bullseye and buster, but because I had to change that in Nextcloud Docker Image, I asume this is the same here?

And the after removing the lines in docker-files.php I would create the image on my synology and create the new image.
Is this correct?

Thanks for the help

**Comment by JC5** [comment]
Sorry I have no clue, I can't support old versions.

**Reply by sn0west** [reply]
okay, thanks for the quick reply. So bad, I realy want to use firefly, because it is simply the best tool out there. I will give it a try.
Can you maybe tel me how to execute the creation of the image? what command and what file I have to kick so that it starts the image creation? Normaly there is a .sh file somewhere but I can't finde any.

**Comment by sn0west** [comment]
I've uploaded the zip file into my nas, somwhere in the folder structure. Unzipped it and changed the dockerfile from bullseye to buster.
then execute this comand 'sudo docker build -t fireflytest . -f Dockerfile'

and got htis as responce.

'ash-4.3# sudo docker build -t firefly3test . -f Dockerfile
Sending build context to Docker daemon  264.7kB
Step 1/43 : FROM debian:buster-slim
 ---> 794cb9f638eb
Step 2/43 : ENV FIREFLY_III_PATH=/var/www/html COMPOSER_ALLOW_SUPERUSER=1 DEBIAN_FRONTEND=noninteractive
 ---> Using cache
 ---> 599a065caa2e
Step 3/43 : ENV BASE_IMAGE_BUILD=40 BASE_IMAGE_DATE="28-10-2023 05:29:11 UTC"
 ---> Using cache
 ---> 0edff9e6f7b3
Step 4/43 : LABEL build_version="4.3" maintainer="james@firefly-iii.org"
 ---> Using cache
 ---> 0fb96c30c00d
Step 5/43 : COPY ../conf/cacert.pem            /etc/ssl/certs/cacert.pem
COPY failed: forbidden path outside the build context: ../conf/cacert.pem ()
'

It seams to start but it has a problem with cacert.pem file or the folder structure. Anyone can help here?
